src/util/psbt/map/output.rs

```diff
@@ -119,31 +119,30 @@ impl TapTree {
         self.0
     }
 
-    /// Returns iterator for a taproot script tree, operating in DFS order over leaf depth and
-    /// leaf script pairs.
-    pub fn iter(&self) -> TapTreeIter {
+    /// Returns [`TapTreeIter`] iterator for a taproot script tree, operating in DFS order over
+    /// tree [`ScriptLeaf`]s.
+    pub fn script_leaves(&self) -> TapTreeIter {
         self.into_iter()
     }
 }
 
 /// Iterator for a taproot script tree, operating in DFS order over leaf depth and
 /// leaf script pairs.
 pub struct TapTreeIter<'tree> {
-    leaf_iter: core::slice::Iter<'tree, LeafInfo>,
+    leaf_iter: core::slice::Iter<'tree, ScriptLeaf>,
 }
 
 impl<'tree> Iterator for TapTreeIter<'tree> {
-    type Item = (u8, &'tree Script);
+    type Item = &'tree ScriptLeaf;
 
+    #[inline]
     fn next(&mut self) -> Option<Self::Item> {
-        self.leaf_iter.next().map(|leaf_info| {
-            (leaf_info.merkle_branch.as_inner().len() as u8, &leaf_info.script)
-        })
+        self.leaf_iter.next()
     }
 }
 
 impl<'tree> IntoIterator for &'tree TapTree {
-    type Item = (u8, &'tree Script);
+    type Item = &'tree ScriptLeaf;
     type IntoIter = TapTreeIter<'tree>;
 
     fn into_iter(self) -> Self::IntoIter {
```

Comment:
Should we really implement `IntoIterator` if we renamed `iter` to `script_leaves`?

Reply:
Added commit removing this implementation.